random_menu_generator.rb

```diff
@@ -0,0 +1,55 @@
+# Random Menu Generator
+# This program takes user input and outputs a chosen number of unique random menu items
+
+# Introduce the generator and ask for input
+puts "Welcome to the Random Menu Generator.\n\n"
+puts "You get to pick a descriptor, a cooking style, and a type of food to combine into a random menu item."
+puts "The generator will make a list of unique menu items, so make sure to enter enough words in each category.\n\n"
+puts "Let's start with adjectives. Please enter a list of adjectives, separated by spaces:"
+
+# Gets input from the user and splits it on white space, then stores that in an array.
+adjectives = gets.chomp.split
+
+# Prompts user to give as many words in the second category as in the first
+puts "\nOkay, now give me some cooking styles.\nYou gave me #{adjectives.length} adjectives, so I recommend that many words here too."
+cooking_styles = gets.chomp.split
+
+# Prompts user to give as many words in the third category as in the smaller of the previous two
+puts "\nOkay, now give me some foods.\nYour smallest set of words so far is #{[adjectives.length, cooking_styles.length].sort[0]}, so I recommend that many words here too."
+foods = gets.chomp.split
+
+# Find length of shortest array
+length_of_shortest = [adjectives.length, cooking_styles.length, foods.length].sort[0]
+
+# Get user input to determine number of menu items displayed
+# Recommend highest limit according to shortest array
+puts "\nHow many menu items would you like to see? (up to #{length_of_shortest})"
+number_of_items = gets.chomp
+puts "\n"
+
+# Validate input to make sure it's a number
+if !(number_of_items.to_i.to_s == number_of_items)
+	puts "That is not a valid number. Please enter a number:"
+	number_of_items = gets.chomp
```

Comment:
Great - I'm glad you see what you could do differently now. I'd agree that the loop until valid is the best option. :+1: